File: 881-boats-to-save-people/881-boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        count=0
        l=0
        r=len(people)-1
        people.sort()
        while l<=r:
            count+=1
            if people[l]+people[r]<=limit:
                l+=1
            r-=1
        return count

File: 881-boats-to-save-people/881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(),people.end());
        int count=0,l=0,r=people.size()-1;
        while(l<=r)
        {
            count++;
            if(people[l]+people[r]<=limit)
            {
                l++;
            }
            r--;
        }
        return count;
    }
};